src/probnum/filtsmooth/statespace/integrator.py

```diff
@@ -54,6 +106,58 @@ def proj2coord(self, coord: int) -> np.ndarray:
         projmat = np.kron(np.eye(self.spatialdim), projmat1d)
         return projmat
 
+    @property
+    def derivwise2coordwise_projmat(self) -> np.ndarray:
+        r"""Projection matrix to change the ordering of the state representation in an :class:`Integrator` from coordinate-wise to derivative-wise representation.
+
+        A coordinate-wise ordering is
+
+        .. math:: (y_1, \dot y_1, \ddot y_1, y_2, \dot y_2, ..., y_d^{(\nu)})
+
+        and a derivative-wise ordering is
+
+        .. math:: (y_1, y_2, ..., y_d, \dot y_1, \dot y_2, ..., \dot y_d, \ddot y_1, ..., y_d^{(\nu)}).
+
+        Default representation in an :class:`Integrator` is coordinate-wise ordering, but sometimes, derivative-wise ordering is more convenient.
+
+        See Also
+        --------
+        :meth:`convert_coordwise_to_derivwise`
+        :meth:`convert_derivwise_to_coordwise`
+
+        """
+        dim = (self.ordint + 1) * self.spatialdim
+        projmat = np.zeros((dim, dim))
+        E = np.eye(dim)
+        for q in range(self.ordint + 1):
+
+            projmat[q :: (self.ordint + 1)] = E[
+                q * self.spatialdim : (q + 1) * self.spatialdim
+            ]
+        return projmat
```

Comment:
I find it easier to think of this reshaping in terms of projection matrices (inverses are transposes). it is a tiny bit slower I suppose, but not used in any performance-reliant code anyway.